ocaml/xapi/xapi_psr_util.ml

```diff
@@ -5,12 +5,22 @@ module Unixext = Stdext.Unixext
 
 let checkpoint_path = "/var/lib/xcp/psr_cp"
 
-let checkpoint_tmp_path = "/var/lib/xcp/psr_cp.tmp"
-
 let old_pool_secret_backup_path = "/var/lib/xcp/ptoken.old"
 
 let new_pool_secret_backup_path = "/var/lib/xcp/ptoken.new"
 
+let read_backups () =
+  try
+    ( old_pool_secret_backup_path
+      |> Unixext.string_of_file
+      |> SecretString.of_string
+    , new_pool_secret_backup_path
+      |> Unixext.string_of_file
+      |> SecretString.of_string )
+  with e ->
+    D.error "xapi_psr_util.ml:read_backups failed" ;
+    raise Api_errors.(Server_error (internal_error, ["failed to read backups"]))
```

Comment:
This does not capture the string for `e` - so it will be difficult to diagnose. You might also want to include the paths for the files.